android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java

```diff
@@ -1,18 +1,210 @@
+
+        questNameEditText = findViewById(R.id.quest_name_edit_text);
+        questImageView = findViewById(R.id.quest_image_view);
+        expEditText = findViewById(R.id.quest_exp_edit_text);
+        statusTextView = findViewById(R.id.quest_status_text_view);
+        statusImageView = findViewById(R.id.quest_status_image_view);
+        viewProofButton = findViewById(R.id.view_proof_button);
+        selectDateButton = findViewById(R.id.quest_select_date_button);
+        selectTimeButton = findViewById(R.id.quest_select_time_button);
+        selectChildSpinner = findViewById(R.id.select_child_spinner);
+        selectRecurrenceSpinner = findViewById(R.id.select_recurrence_spinner);
+        descriptionEditText = findViewById(R.id.description_edit_text);
+        cancelButton = findViewById(R.id.cancel_button);
+        saveButton = findViewById(R.id.save_button);
+
+        // TODO set all initial values to actual quest info
+        mYear = 1996;
+        mMonth = 10;
+        mDay = 29;
+        mHour = 5;
+        mMinute = 30;
+
+        // Quest Name
+        // TODO: fetch actual value
+        questName = "Sweep kitchen floor";
+        questNameEditText.setText(questName);
+
+        // Quest icon
+        // TODO: fetch quest icon based on name
+        int icon = R.drawable.cleaner;
+        questImageView.setImageResource(icon);
+
+        // Quest Exp
+        // TODO: fetch actual value
+        expEditText.setText("20");
+
+        // status
+        // TODO: fetch actual value and possibly hide button
+        int statusText = R.string.pending_approval;
+        statusTextView.setText(getString(statusText));
+        int statusImage;
+        if (statusText == R.string.pending_approval) {
+            statusImage = R.drawable.yellow_check_mark;
+        } else if (statusText == R.string.completed) {
+            statusImage = R.drawable.green_check_mark;
+        } else {
+            viewProofButton.setVisibility(viewProofButton.GONE);
+            statusImage = R.drawable.red_check_mark;
+        }
+        statusImageView.setImageResource(statusImage);
+
+        // Select Date
+        selectDateButton.setText(getDateText());
+        selectDateButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                disableButtons(); // ?
+                final Calendar c = Calendar.getInstance();
+                int year = c.get(Calendar.YEAR);
+                int month = c.get(Calendar.MONTH);
+                int day = c.get(Calendar.DAY_OF_MONTH);
+
+                DatePickerDialog.OnDateSetListener listener = new DatePickerDialog.OnDateSetListener() {
+                    @Override
+                    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+                        mDay = dayOfMonth;
+                        mMonth = month;
```

Comment:
month is represented by an integer from 0 to 11
do `mMonth = month + 1` here instead of line 203
to reduce the number of places you need to do this adjustment as you still need to make a request with mMonth

Reply:
I want to keep mMonth as it is so I have the correct value when I want to use it to send to server

Reply:
nvm changed it